src/Compiler/Driver/CompilerOptions.fs

```diff
@@ -2383,12 +2393,30 @@ let ReportTime (tcConfig: TcConfig) descr =
 
         let tStart =
             match tPrev, nPrev with
-            | Some (tStart, tPrev, utPrev, gcPrev), Some (prevDescr, _) ->
+            | Some (tStart, tPrev, utPrev, gcPrev), Some (prevDescr, prevActivity) ->
                 let spanGC = [| for i in 0..maxGen -> GC.CollectionCount i - gcPrev[i] |]
                 let t = tNow - tStart
                 let tDelta = tNow - tPrev
                 let utDelta = utNow - utPrev
 
+                match prevActivity with
+                | :? System.Diagnostics.Activity as a when isNotNull a ->
+                    // Yes, there is duplicity of code between the console reporting and Activity collection right now.
+                    // If current --times behaviour can be changed (=breaking change to the layout etc.), the GC and CPU time collecting logic can move to Activity
```

Comment:
You could argue that it's possible to retain the old console formatting exactly as is, but only changing the underlying source of data - but maybe that's unnecessary work.

Reply:
Yep, exactly - it is possible, but would mean dropping a more generalized approach and hardcoding the layout to fit what use to be there.